cluster-autoscaler/cloudprovider/cloudifyprovider/scale_provider.go

```diff
@@ -57,69 +117,126 @@ func (clsp *CloudifyScaleProvider) Refresh() error {
 	}
 
-	if deployment.ScalingGroups != nil {
-		for groupName := range deployment.ScalingGroups {
-			nodes = append(nodes, NodeToNodeGroup(clsp.client, clsp.deploymentID, groupName))
-		}
+	if items == nil {
+		glog.Errorf("no deployments found !!!")
 	}
 
+	for _, dep := range items {
+		glog.V(4).Infof("NodeGroups, Deployment: %+v", dep["id"])
+		deployment, err := clsp.client.GetDeployment(dep["id"])
+		if err != nil {
+			glog.Errorf("Cloudify error: %s\n", err.Error())
+			return nodes
+		}
+
+		if deployment.ScalingGroups != nil {
+			for groupName := range deployment.ScalingGroups {
+				nodes = append(nodes, NodeToNodeGroup(clsp.client, dep["id"], groupName, dep["node_data_type"]))
+			}
+		}
+	}
 	glog.Warningf("NodeGroups: %+v", nodes)
 	return nodes
 }
 
 // NodeGroupForNode returns the node group for the given node.
 func (clsp *CloudifyScaleProvider) NodeGroupForNode(kubeNode *apiv1.Node) (cloudprovider.NodeGroup, error) {
-	glog.V(4).Infof("NodeGroupForNode(%v.%v)", clsp.deploymentID, kubeNode.Name)
-
-	groupedInstances, err := clsp.client.GetDeploymentInstancesScaleGrouped(
-		clsp.deploymentID, cloudify.KubernetesNode)
-
+	items, err := clsp.GetDeploymentNodeInfo()
 	if err != nil {
 		glog.Errorf("Cloudify error: %s\n", err.Error())
-		return nil, err
 	}
 
-	for groupName, cloudInstances := range groupedInstances {
-		// search instance
-		for _, cloudInstance := range cloudInstances.Items {
-			// check runtime properties
-			if cloudInstance.GetStringProperty("hostname") != kubeNode.Name {
-				continue
+	if items == nil {
+		glog.Errorf("no deployments found !!!")
+	}
+	for _, dep := range items {
+		if dep["deployment_type"] == "node" {
+			glog.V(4).Infof("NodeGroupForNode(%v.%v)", dep["id"], kubeNode.Name)
+			groupedInstances, err := clsp.client.GetDeploymentInstancesScaleGrouped(
+				dep["id"], dep["node_data_type"])
+
+			if err != nil {
+				glog.Errorf("Cloudify error: %s\n", err.Error())
+				return nil, err
 			}
 
-			return NodeToNodeGroup(clsp.client, clsp.deploymentID, groupName), nil
+			for groupName, cloudInstances := range groupedInstances {
+				// search instance
+				for _, cloudInstance := range cloudInstances.Items {
+					// check runtime properties
+					if cloudInstance.GetStringProperty("hostname") != kubeNode.Name {
+						continue
+					}
+
+					return NodeToNodeGroup(clsp.client, dep["id"], groupName, dep["node_data_type"]), nil
+				}
+			}
+			glog.Warningf("NodeGroupForNode(%v.%v): Skiped, check list of members in scale group", dep["id"], kubeNode.Name)
 		}
 	}
-	glog.Warningf("NodeGroupForNode(%v.%v): Skiped, check list of members in scale group", clsp.deploymentID, kubeNode.Name)
+
 	return nil, nil
 }
 
 // Pricing returns pricing model for this cloud provider or error if not available.
 func (clsp *CloudifyScaleProvider) Pricing() (cloudprovider.PricingModel, errors.AutoscalerError) {
-	glog.Errorf("?Pricing: %v", clsp.deploymentID)
+	items, err := clsp.GetDeploymentNodeInfo()
```

Comment:
Looks as debug code, could we drop it?

Reply:
We don't need it since I change how we handle deployment and no need to have it here
@0lvin

Reply:
@mabuaisha So remove :-)